File: LeetCode/366-find-leaves-of-binary-tree/ex_366.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> findLeaves(TreeNode* root) {
        vector<vector<int>> res;
        height(res, root);
        return res;
    }
private:
    int height(vector<vector<int>> &res, TreeNode* root) {
        if (root == NULL)   return -1;
        int level = 1 + max(height(res, root->left), height(res, root->right));
        if (res.size() < level + 1)     res.push_back(vector<int>());
        res[level].push_back(root->val);
        return level;
    }
};

File: LeetCode/366-find-leaves-of-binary-tree/ex_366.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def findLeaves(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        res = []
        self.height(res, root)
        return res
    
    def height(self, res, root):
        if not root:    return -1
        level = 1 + max(self.height(res, root.left), self.height(res, root.right))
        if len(res) < level + 1:    res.append([])
        res[level].append(root.val)
        return level
            